### DISCUSSION TITLE
Reason for 1,1 at top left?

### DISCUSSION THREAD

**Discussion by rduke1982** [selimanac/defold-astar#6] [General]
Considering Defold uses 1,1 as the bottom left, what was the reason for making it the top left in the extension? Or is my understanding wrong?

**Comment by selimanac** [comment]
Map is essentially a single-dimensional array, and every array starts at 0 (or 1 in Lua). It’s up to you how to set up the map—you may choose to use the built-in tilemap editor or not. If needed, you can use `astar.map_vflip() ` (or you can vflip on setup)for vertical flipping.

**Comment by rduke1982** [comment]
<img width="796" alt="Screenshot 2024-12-08 at 5 05 48 PM" src="https://github.com/user-attachments/assets/3a292943-df97-4784-b0f3-18e14c161239">

I think I understand, but I want to make sure. This isn't a major issue, I'm just curious.

In Defold 1,1 is the red circle. But in Astar, the array is 1,-1 in Defold — the blue circle.

Is my understanding correct? 

vflip wouldn't work because it would still have the same -y coords.

**Reply by rduke1982** [reply]
I guess vflip would work. It's confusing because of how the print looks, but it makes sense otherwise.

<img width="717" alt="image" src="https://github.com/user-attachments/assets/7fc08341-bd65-4ee3-b840-bee19b627904">

Without flipping, clicking tile 3,10 would be 3,4. But after flipping, it matches the correct tile in the astar array.

**Reply by selimanac** [reply]
I'm quite tired today, so I guess I might be misleading you. Sorry about that! Here is a very basic example using built-in tilemaps:
[astar_test.zip](https://github.com/user-attachments/files/18052666/astar_test.zip)
You can also check out older examples here:
https://github.com/selimanac/defold-astar-examples/tree/master

**Reply by rduke1982** [reply]
I got it working how I want, for now. Thanks for the help! I shared a short video of [my result](https://discord.com/channels/250018174974689280/250018669512359936/1315331148225970248) in Discord.
